hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/ssl/PemFileBasedKeyStoresFactory.java

```diff
@@ -44,87 +45,48 @@
  */
 @InterfaceAudience.Private
 @InterfaceStability.Evolving
-public class PemFileBasedKeyStoresFactory implements KeyStoresFactory {
+public class PemFileBasedKeyStoresFactory implements KeyStoresFactory,
+    CertificateNotification {
 
   private static final Logger LOG =
       LoggerFactory.getLogger(PemFileBasedKeyStoresFactory.class);
 
-  /**
-   * The name of the timer thread monitoring file changes.
-   */
-  public static final String SSL_MONITORING_THREAD_NAME =
-      "SSL Certificates Store Monitor";
-
   /**
    * Default format of the keystore files.
    */
   public static final String DEFAULT_KEYSTORE_TYPE = "jks";
 
-  private KeyManager[] keyManagers;
-  private TrustManager[] trustManagers;
-  private Timer monitoringTimer;
+  private AtomicReference<KeyManager[]> keyManagers;
+  private AtomicReference<TrustManager[]> trustManagers;
```

Comment:
Is there a reason to use an AtomicReference to the arrays? As I see all access is synchronized, while we even give out the reference to the array from the class via the getter.

I would suggest two things regarding these arrays:
- as all access to these via the getter accesses the 0th element, for now we can expose a method that returns the 0th element, and if we need more, we can change. 
- It might worth to revisit if we need these to be arrays, in the current implementation we seem to always use the 0th element, and we are storing these as arrays because underlying APIs return an array.

Reply:
@fapifta , thanks for the review.

AtomicReference is not a must to have. I will revert it in next new patch.  As for the array type, it's because the underlying APIs return and array.